src/js/components/AppHealthBarWithTooltipComponent.jsx

```diff
@@ -1,84 +1,61 @@
   displayName: "AppHealthBarWithTooltipComponent",
 
-  mixins: [TooltipMixin],
-
   propTypes: {
     model: React.PropTypes.object.isRequired
   },
 
   getInitialState: function () {
     return {
-      tipContent: this.getAppHealthBreakdown(this.props.model)
+      isPopoverVisible: false
     };
   },
 
   shouldComponentUpdate: function (nextProps, nextState) {
-    // Ensure that the health bar does not update before the tipContent was
-    // updated (`setState` is not guaranteed to complete before `render`)
-    if (this.state.tipContent !== nextState.tipContent) {
-      return true;
-    }
-    var healthWasUpdated =
+    return  this.state.isPopoverVisible !== nextState.isPopoverVisible ||
       !Util.compareArrays(this.props.model.health, nextProps.model.health);
-    if (healthWasUpdated) {
-      this.setState({
-        tipContent: this.getAppHealthBreakdown(nextProps.model)
-      });
-    }
-    return false;
   },
 
-  handleMouseOverHealthBar: function () {
-    var el = this.refs.healthBar.getDOMNode();
-    this.tip_showTip(el);
+  handleMouseOverHealthBar: function (e) {
+    this.setState({
+      isPopoverVisible: true
+    });
   },
 
   handleMouseOutHealthBar: function () {
-    var el = this.refs.healthBar.getDOMNode();
-    this.tip_hideTip(el);
-  },
-
-  getAppHealthBreakdown: function (model) {
-    let component = (
-      <AppHealthDetailComponent
-        className="list-unstyled"
-        fields={[
-          HealthStatus.HEALTHY,
-          HealthStatus.UNHEALTHY,
-          HealthStatus.UNKNOWN,
-          HealthStatus.STAGED,
-          HealthStatus.OVERCAPACITY,
-          HealthStatus.UNSCHEDULED
-        ]}
-        model={model} />
-    );
-    return React.renderToString(component);
+    this.setState({
+      isPopoverVisible: false
+    });
   },
 
   render: function () {
-    // Extra <div> nesting is necessary here because of the way the TooltipMixin
-    // searches for elements - it finds children
+    var model = this.props.model;
+
     return (
-      <div>
-        <div ref="healthBar"
-          data-behavior="show-tip"
-          data-tip-type-class="default"
-          data-tip-place="top-left"
-          data-tip-content={this.state.tipContent}
-          onMouseOver={this.handleMouseOverHealthBar}
-          onMouseOut={this.handleMouseOutHealthBar}>
-          <AppHealthBarComponent
-            model={this.props.model}/>
-        </div>
+      <div onMouseOver={this.handleMouseOverHealthBar}
+           onMouseOut={this.handleMouseOutHealthBar}>
+        <PopoverComponent visible={this.state.isPopoverVisible}>
+          <AppHealthDetailComponent
+            className="list-unstyled"
+            fields={[
+              HealthStatus.HEALTHY,
+              HealthStatus.UNHEALTHY,
+              HealthStatus.UNKNOWN,
+              HealthStatus.STAGED,
+              HealthStatus.OVERCAPACITY,
+              HealthStatus.UNSCHEDULED
+            ]}
```

Comment:
We would lose the field order with the `Object.values` approach. 	Dunno if the order is important. 

/cc @philipnrmn